### PR TITLE
feat(discover) Add top5 results mode to graph

### PR DESCRIPTION
**Pull request by markstory**
Add a new top5 display mode to the discover results graph. I've done a bunch of documentation shuffling and reduced the amount of redundant data we generate when preparing multi-series results.

Screen Shot 2020-04-16 at 4 36 37 PM

### CODE REVIEW

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx [hunk 1]
```diff
@@ -29,38 +29,115 @@ export type TimeSeriesData = {
...
   keyTransactions?: boolean;
+  /**
+   * The number of top results to get. When set a multi-series result will be returned
+   * in the `results` child render function.
+   */
+  topEvents?: number;
+  /**
+   * How to order results when getting top events.
+   */
+  orderby?: string;
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx:140]: Thanks for moving these. They're a lot more useful on the types.

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx [hunk 2]
```diff
@@ -327,11 +338,10 @@ class EventsRequest extends React.PureComponent<EventsRequestProps, EventsReques
   /**
    * Transforms query response into timeseries data to be used in a chart
    */
-  transformTimeseriesData(data: EventsStatsData): Series[] {
-    const seriesName = this.props.currentSeriesName ?? 'Current';
+  transformTimeseriesData(data: EventsStatsData, seriesName?: string): Series[] {
     return [
       {
-        seriesName,
+        seriesName: seriesName || 'Current',
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx:344]: Why drop null coalescing operator?

**Review by dashed — APPROVED** [review]

**Review by matejminar — APPROVED** [review]
Nice 🚀

File: src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/events/utils/eventsRequest.tsx:344]: I wanted to squash away 0 and '' as bad values too.

**Pull request merged**